components/Chips/tests/unit/ChipsTests.m

```diff
@@ -394,4 +397,90 @@ - (void)testChipsWithDeleteEnabled {
+  chipView.mdc_adjustsFontForContentSizeCategory = YES;
+  chipView.mdc_legacyFontScaling = YES;
+  chipView.titleLabel.text = @"Chip";
+  [chipView updateTitleFont];
+  CGFloat defaultFontSize = chipView.titleLabel.font.pointSize;
+
+  // When
+  if (@available(iOS 10.0, *)) {
+    Class class = object_getClass(UIScreen.mainScreen.traitCollection);
+    SEL originalSelector = @selector(preferredContentSizeCategory);
+    SEL swizzledSelector = @selector(swizzled_preferredContentSizeCategory);
+    original_TraitCollection_Method = class_getInstanceMethod(class, originalSelector);
+    swizzled_TraitCollection_Method = class_getInstanceMethod([self class], swizzledSelector);
+    method_exchangeImplementations(original_TraitCollection_Method,
+                                   swizzled_TraitCollection_Method);
+  }
+  [chipView updateTitleFont];
+  CGFloat fontSizeForExtraExtraLargeSizeCategory = chipView.titleLabel.font.pointSize;
+
+  // Then
+  if (@available(iOS 10.0, *)) {
+    XCTAssertGreaterThan(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
+  } else {
+    XCTAssertEqual(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
+  }
+  [self swizzled_teardown];
+}
+
+- (void)testChipViewDynamicTypeBehavior {
+  // Given
+  MDCChipView *chipView = [[MDCChipView alloc] init];
+  UIFont *titleFont = [UIFont systemFontOfSize:14.0 weight:UIFontWeightMedium];
+  MDCFontScaler *fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody2];
+  titleFont = [fontScaler scaledFontWithFont:titleFont];
+  titleFont = [titleFont mdc_scaledFontAtDefaultSize];
+  chipView.titleFont = titleFont;
+  chipView.mdc_adjustsFontForContentSizeCategory = YES;
+  chipView.titleLabel.text = @"Chip";
+  [chipView updateTitleFont];
+  CGFloat defaultFontSize = chipView.titleLabel.font.pointSize;
+
+  // When
+  if (@available(iOS 10.0, *)) {
+    Class class = object_getClass(UIScreen.mainScreen.traitCollection);
+    SEL originalSelector = @selector(preferredContentSizeCategory);
+    SEL swizzledSelector = @selector(swizzled_preferredContentSizeCategory);
+    original_TraitCollection_Method = class_getInstanceMethod(class, originalSelector);
+    swizzled_TraitCollection_Method = class_getInstanceMethod([self class], swizzledSelector);
+    method_exchangeImplementations(original_TraitCollection_Method,
+                                   swizzled_TraitCollection_Method);
+  }
+  [chipView updateTitleFont];
+  CGFloat fontSizeForExtraExtraLargeSizeCategory = chipView.titleLabel.font.pointSize;
+
+  // Then
+  if (@available(iOS 10.0, *)) {
+    XCTAssertGreaterThan(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
+  } else {
+    XCTAssertEqual(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
+  }
+  [self swizzled_teardown];
+}
+
+- (id)swizzled_preferredContentSizeCategory {
+  return UIContentSizeCategoryExtraExtraLarge;
+}
+
+- (void)swizzled_teardown {
```

Comment:
I know we talked about using swizzling in tests yesterday in a meeting, but it was as a last resort. I'm not against it, and I don't want to block you on this PR, but I feel like we should be careful in how we introduce, and come up with a standard way of doing it.

Reply:
Yea. I prefer doing injection actually, but it would require a relative large refactor (including creating wrapper) in Chips and traitCollection potentially. I see swizzling here as an easy win for validating the functionality. We can or should revisit it sometime during our meeting. I don't felt it is risky.

Reply:
I'm not opposed to swizzling, I just feel like we should rein it in and put it behind some APIs that all the tests can make use of, rather than have a bunch of independent instances of swizzling throughout our tests. I guess we'll figure that out later though.

Reply:
Im moving some of this conversation to internal chat because i am going to reference some internal code.